runtime/interpreter-gen-x64.cpp

```diff
@@ -996,6 +996,76 @@ void emitHandler<BINARY_SUBSCR_LIST>(EmitEnv* env) {
   emitHandleContinue(env, kGenericHandler);
 }
 
+void emitHeaderCountOrOverflow(EmitEnv* env, Register r_dst,
+                               Register r_container) {
+  // Load header().count() as a SmallInt
+  // r_dst = header().count()
+  __ movq(r_dst,
+          Address(r_container, heapObjectDisp(RawHeapObject::kHeaderOffset)));
+  // TODO(emacs): Does this need to be sarq like emitConvertFromSmallInt?
+  __ shrq(r_dst, Immediate(RawHeader::kCountOffset - Object::kSmallIntTagBits));
+  __ andq(r_dst, Immediate(Header::kCountMask << Object::kSmallIntTagBits));
+  // if (r_dst == kCountOverflowFlag)
+  __ cmpq(r_dst, smallIntImmediate(RawHeader::kCountOverflowFlag));
+  Label done;
+  // TODO(emacs): Use cmov?
```

Comment:
I think no because the branch prediction should be pretty decent here and we don't want the data dependency. That being said, I haven't actually tested timing on this. Going to keep as is for now.